### ISSUE TITLE
ansible-lint give no output and just a python error

### ISSUE THREAD

**Issue by imranshaik5155** [ansible/ansible-lint#850]
##### Summary
I am running the ansible-lint on all of our ansible code with 500+ roles, and it is showing no output and failing with python error.
How can I find where it is actually failing and with what reason.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible 2.9.9
  config file = None
  configured module search path = ['/Users/imranshaik/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.6 (default, Dec 30 2019, 19:38:26) [Clang 11.0.0 (clang-1100.0.33.16)]

ansible-lint --version
ansible-lint 4.2.0

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
mac OS

```python-traceback
Traceback (most recent call last):
  File "/usr/local/bin/ansible-lint", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/__main__.py", line 187, in main
    matches.extend(runner.run())
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/__init__.py", line 287, in run
    skip_list=self.skip_list))
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/__init__.py", line 178, in run
    matches.extend(rule.matchyaml(playbookfile, text))
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/__init__.py", line 119, in matchyaml
    yaml = ansiblelint.utils.append_skipped_rules(yaml, text, file['type'])
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/utils.py", line 596, in append_skipped_rules
    yaml_skip = _append_skipped_rules(pyyaml_data, file_text, file_type)
  File "/usr/local/lib/python3.7/site-packages/ansiblelint/utils.py", line 635, in _append_skipped_rules
    if pyyaml_task.get('name') != ruamel_task.get('name'):
AttributeError: 'str' object has no attribute 'get'
```

**Comment by ssbarnea** [comment]
Bug form asks for using latest version from pypi or master, which are 4.3.x ones. Please use them and you are unlikely to get the same error.

**Comment by imranshaik5155** [comment]
@ssbarnea I have upgraded to the latest version

```
/usr/local/bin/ansible-lint --version
ansible-lint 4.3.0a2.dev66+ga9d0ca1
```
but still I am getting the same issue. please help

**Comment by ssbarnea** [comment]
@imranshaik5155 Run in verbose mode to identify the file causing the failure. If is not an Ansible file you may want to add it to the skiplist. Either way, look inside the file, try to reduce it until you can find a way to reproduce the bug.

I will reopen the bug as current exception is far from friendly but without a sample file nobody will be able to help.

**Comment by nealep** [comment]
I get the same error if I have the following in a tasks/main.yml file in a role: 

```yaml
---
# tasks file for role
tasks:
```

**Comment by webknjaz** [comment]
@nealep plz post your traceback. We need to compare it. Or better post the whole verbose run from master.

**Comment by nealep** [comment]
Here's the output using the dummy file from before: 

```shell
username@DESKTOP-FQ0MMHQ: pip3 install git+https://github.com/ansible/ansible-lint.git
username@DESKTOP-FQ0MMHQ:~/ansible-slurm$ ansible-lint -vv ./roles/munge_client/
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(colored=True, config_file=None, display_relative_path=True, exclude_paths=[], format='plain', listrules=False, listtags=False, parseable=False, parseable_severity=False, playbook=['./roles/munge_client/'], quiet=False, rulesdir=[], skip_list=[], tags=[], use_default_rules=False, verbosity=2)
DEBUG    Loading rules from /home/username/.local/lib/python3.6/site-packages/ansiblelint/rules
DEBUG    Examining roles/munge_client/tasks/main.yml of type tasks
Traceback (most recent call last):
  File "/home/username/.local/bin/ansible-lint", line 11, in <module>
    load_entry_point('ansible-lint==0.0.0', 'console_scripts', 'ansible-lint')()
  File "/home/username/.local/lib/python3.6/site-packages/ansiblelint/__main__.py", line 116, in main
    matches.extend(runner.run())
  File "/home/username/.local/lib/python3.6/site-packages/ansiblelint/runner.py", line 95, in run
    skip_list=self.skip_list))
  File "/home/username/.local/lib/python3.6/site-packages/ansiblelint/rules/__init__.py", line 222, in run
    matches.extend(rule.matchtasks(playbookfile, text))
  File "/home/username/.local/lib/python3.6/site-packages/ansiblelint/rules/__init__.py", line 85, in matchtasks
    yaml = append_skipped_rules(yaml, text, file['type'])
  File "/home/username/.local/lib/python3.6/site-packages/ansiblelint/skip_utils.py", line 68, in append_skipped_rules
    yaml_skip = _append_skipped_rules(pyyaml_data, file_text, file_type)
  File "/home/username/.local/lib/python3.6/site-packages/ansiblelint/skip_utils.py", line 119, in _append_skipped_rules
    if pyyaml_task.get('name') != ruamel_task.get('name'):
AttributeError: 'AnsibleUnicode' object has no attribute 'get'
```

**Comment by ssbarnea** [comment]
While this construct is valid yaml, it assigns null to tasks list. I guess we need a new rule AvoidNullAnsibleBlocks or a similar name.

Avoiding that stacktrace is easy. Do you want to propose a new rule?

**Comment by nealep** [comment]
I'm not sure how to put in new rules. What would I need to do to add one?

**Comment by ssbarnea** [comment]
I already have some WIP done but I am waiting for the 2.10 changes to merge as I need them to pass tests. I will make a PR with it right after that. That rule is bit more complex then the normal ones as it involves fixing a parsing error.
